Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp

```diff
@@ -99,11 +99,21 @@ WebIDL::ExceptionOr<void> CSSStyleSheet::remove_rule(unsigned index)
 
 void CSSStyleSheet::for_each_effective_style_rule(Function<void(CSSStyleRule const&)> const& callback) const
 {
+    for (auto& media : m_media_query_list) {
+        if (!media.matches()) {
+            return;
+        }
+    }
+
     m_rules->for_each_effective_style_rule(callback);
 }
 
 bool CSSStyleSheet::evaluate_media_queries(HTML::Window const& window)
 {
+    for (auto& media : m_media_query_list) {
+        media.evaluate(window);
+    }
+
     return m_rules->evaluate_media_queries(window);
```

Comment:
My mistake for not marking `MediaQuery::evaluate()` as `[[nodiscard]]`, but if any of them return `true` then we need to make sure that `evaluate_media_queries()` also returns true. Something like this:

```suggestion
    bool any_media_queries_changed_match_state = false;

    for (auto& media : m_media_query_list) {
        if (media.evaluate(window))
            any_media_queries_changed_match_state = true;
    }
    
    if (m_rules->evaluate_media_queries(window))
    	any_media_queries_changed_match_state = true;

    return any_media_queries_changed_match_state;
```

I haven't tested this code but hopefully you get the idea - we need to evaluate all media queries, and return true if any one of them did change.